### ISSUE TITLE
Narrow and misaligned boxes in plotly express box plot when using color= 

### ISSUE THREAD

**Issue by rsgoodwin** [plotly/plotly.py#2708]
When using the same variable (column) in a dataframe for both the x and color variable in a plotly express chart,  the boxes become narrow and no longer centered over the x axis labels.  

Repeatable code below.   Without the color='x',  the boxes are of appropriate width and centered on the x axis labels as desired. 

```
import pandas as pd
import numpy as np
import plotly.express as px

df = pd.DataFrame({
    'x': np.random.choice(list('ABCDE'), 100),
    'y': np.random.normal(10, 1, 100)
})

b1 = px.box(
    data_frame = df,
    x = 'x',
    y = 'y',
    color = 'x'
)

b1.show()
```

**Comment by nicolaskruchten** [comment]
Thanks for the nice reproducible example! This bug was fixed in version 4.8 I believe, and I recommend upgrading to version 4.9 if you can, although 4.10 should be out within the next 10 days :)

**Comment by rsgoodwin** [comment]
Thanks for the quick reply!  And... shoot,  I forgot I was using one of our work envs that probably had 4.6.  Re-running on my personal laptop that has 4.9,  and yes,  bar width and alignment is perfect.  Merci encore!
